views.go

```diff
@@ -434,13 +436,18 @@ func (lk *logKeeper) viewBucketBuild(r *http.Request, buildID string) (*model.Bu
 	}()
 	wg.Wait()
 
+	if pail.IsKeyNotFoundError(buildErr) {
+		if shouldFallBack {
```

Comment:
My understanding is that at the time we'll be turning this on all builds will be in the db and a subset of builds will be in S3. 
If so, could it make sense that when the feature flag indicates that we should read from S3 we 1) check the db to make sure it's in S3 and if not 2) fallback to the db. This way requests for db logs won't need to wait on the S3 responses (assuming db queries are in-fact faster)